frontend/lib/src/features/password_reset/forgot_password_page.dart

```diff
@@ -0,0 +1,169 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_form_builder/flutter_form_builder.dart';
+import 'package:form_builder_validators/form_builder_validators.dart';
+import 'package:get/get.dart';
+
+import 'package:frontend/src/controllers/button_controller.dart';
+import 'package:frontend/src/features/login/login_page.dart';
+import 'package:frontend/src/components/auth/auth_header.dart';
+import 'package:frontend/src/components/input/input_field.dart';
+import 'package:frontend/src/components/button.dart';
+import 'package:frontend/src/controllers/login_controller.dart';
+import 'package:frontend/src/features/password_reset/components/reset_password_page.dart';
+import 'package:frontend/src/helper/dialog/show_alert_dialog.dart';
+import 'package:frontend/src/helper/dio.dart';
+
+/*
+  Page for the Forgot Password
+  User inputs email
+*/
+class ForgotPasswordPage extends StatelessWidget {
+  const ForgotPasswordPage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final ButtonController buttonController = Get.put(ButtonController());
+    final formKey = GlobalKey<FormBuilderState>();
+
+    void onSubmit() async {
+      // disable the button to prevent multiple requests being sent
+      buttonController.setButtonEnabled = false;
+
+      final client = NetworkConfig().client;
+
+      final email = formKey.currentState!.fields['email']!.value;
+
+      final forgotPasswordResponse = await client.post(
+        forgotPasswordUrl,
+        data: {
+          'email': email,
+        },
+      );
+
+      if (forgotPasswordResponse.statusCode == HttpStatus.ok) {
+        final message = forgotPasswordResponse.data['data']['message'];
+
+        showAlertDialog(
+          title: 'SUCCESS',
+          content: message,
+          onClick: () {
+            Get.to(ResetPasswordPage(emailValue: email));
+          },
+        );
+      } else if (forgotPasswordResponse.statusCode == HttpStatus.badRequest) {
+        // the error response is in Response<dynamic>, toString + jsonDecode to easily access data
+        final error = jsonDecode(forgotPasswordResponse.data.toString());
+        final message = error['error']['message']['email'][0];
+
+        if (message == "The email you entered isn't connected to an account.") {
+          formKey.currentState?.fields['email']?.invalidate(message);
+        }
+
+        final throttledMessage = error['error']['message']['short'][0];
```

Comment:
```suggestion
        final throttledMessage = error['error']['message']['short']?[0];
```
add null checks here, 
to check, in be change this key to "short2"
not sure if just adding "?" is enough